### ISSUE TITLE
How to improve the accuracy in night？

### ISSUE THREAD

**Issue by wcg5262** [cardwing/Codes-for-Lane-Detection#19]
Now I have a project that  requires  lane detection accuracy of 80% in night， If I use SCNN or LaneNet to train the data set in night alone, can I reach 80 percent?

**Comment by cardwing** [comment]
@wcg5262, if you need the model to achieve 80% accuracy at night in real-world applications, the best way is to directly train the model in that road conditions. If you just want to achieve 80% accuracy in one specific dataset, you need to collect more video frames in night conditions and train the model using these frames. You can have a try to first train SCNN or LaneNet using all frames in CULane and then fine tune the model only using frames at night.

**Comment by wcg5262** [comment]
> 
> 
> @wcg5262, if you need the model to achieve 80% accuracy at night in real-world applications, the best way is to directly train the model in that road conditions. If you just want to achieve 80% accuracy in one specific dataset, you need to collect more video frames in night conditions and train the model using these frames. You can have a try to first train SCNN or LaneNet using all frames in CULane and then fine tune the model only using frames at night.



Now I use LaneNet to train some pictures of tusimple and some pictures of daytime and dark days in culane, but the effect is not good. The weight test of tusimple is good, but the test of culane's pictures is bad. Can you give some guidance for this reason? How do I train now?

**Comment by wcg5262** [comment]
@cardwing OK，thanks for your patient answer

**Comment by wcg5262** [comment]
@cardwing oh~ . I would also like to ask if I use LaneNet to train culane alone，Should I resize the picture to 512 x 256 or 800×288 ？

**Comment by cardwing** [comment]
@wcg5262, 288 x 800 is used for CULane and 256 x 512 is used for TuSimple.

**Comment by wcg5262** [comment]
@cardwing ok

**Comment by wcg5262** [comment]
@cardwing I just ran the test from your code， but the output is binary image about every lane，how can I use it in original image to achieve lane detection.

**Comment by cardwing** [comment]
Now, you get the probability maps from our model. To get the final performance, you need to follow [SCNN](https://github.com/XingangPan/SCNN) to get curve lines from probability maps as well as calculate precision, recall and F1-measure. Just follow the instructions in README.

**Comment by wcg5262** [comment]
@cardwing Hello， I just found that both the model trained by LaneNet authors and the model trained by myself seem to be over-fitting. It seems difficult for me to solve this problem now. I found you added the dropout to the code and LaneNet didn't. Did you have any fitting problems when training SCNN models? Where should I add dropout layer to LaneNet?

**Comment by cardwing** [comment]
@wcg5262, you can do some data augmentation like horizontal flipping and random cropping. Besides, you can add dropout layers before FCN of LaneNet (see SCNN-Tensorflow for details). SCNN-Tensorflow should outperform LaneNet since it has message passing modules. You can try SCNN-Tensorflow and check if it can satisfy your requirement.

**Comment by wcg5262** [comment]
@cardwing Thanks for your advise. Now I try to finish my project with lanenet as much as possible, because it may take a long time to learn SCNN algorithm and code from scratch, and then to do data sets. I'm in a  little hurry now.
